cli/internal/globby/globby.go

```diff
@@ -101,7 +101,10 @@ func globWalkFs(fsys iofs.FS, fsysRoot string, basePath string, includePatterns
 		// This will not error as it follows the call to checkRelativePath.
 		iofsRelativePath, _ := fs.IofsRelativePath(fsysRoot, excludePath)
 
-		// Excludes operate on entire folders.
+		// In case this is a file pattern and not a directory, add the exact pattern.
+		processedExcludes = append(processedExcludes, iofsRelativePath)
+		// TODO: we need to either document or change this behavior
```

Comment:
I'm not a fan of `"outputs": ["glob-syntax-1", "!glob-syntax-2"]`. We're not using `.gitignore` syntax for inclusions, so I don't think we should use it for exclusions either.

Reply:
I'm 100% on board with aligning the syntax, I was pointing out that we could elect to align in the _other_ direction, which would have the bonus feature of being (likely) backwards compatible.

So that would result in these two being identical:
- `includes: "foo"` (where foo is a dir) 
- `includes: "foo/**"`

Unfortunately we didn't really design ourselves into our current state and attaching a design now is much more complicated.

Reply:
> attaching a design now is much more complicated.

Definitely agreed

Supporting `.gitignore` (and/or others? `.dockerignore`?) is an interesting idea. If it's breaking, we might need to wait for 2.0. If it's not, that gives us more flexibility....